nixos/modules/services/misc/llama-cpp.nix

```
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.llama-cpp;
+in {
+
+  options = {
+
+    services.llama-cpp = {
+      enable = lib.mkEnableOption (lib.mdDoc "LLaMA C++ server");
+
+      package = lib.mkPackageOption pkgs "llama-cpp" { };
+
+      model = lib.mkOption {
+        type = lib.types.path;
+        example = "/models/mistral-instruct-7b/ggml-model-q4_0.gguf";
+        description = lib.mdDoc "Model path.";
+      };
+
+      extraFlags = lib.mkOption {
+        type = lib.types.str;
+        description = lib.mdDoc "Extra flags passed to llama-cpp-server.";
+        example = "-c 4096 -ngl 32 --numa";
+        default = "";
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        description = lib.mdDoc "IP address the LLaMA C++ server listens on.";
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 8080;
+        description = lib.mdDoc "Listen port for LLaMA C++ server.";
+      };
+
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Open ports in the firewall for LLaMA C++ server.
+        '';
+      };
+    };
+
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    systemd.services.llama-cpp = {
+      description = "LLaMA C++ server";
+      after = ["network-online.target"];
```

This is usually enough for servers.  See `man systemd.special`.

```suggestion
      after = ["network.target"];
```